api_shield_schemas.go

```diff
@@ -333,3 +333,114 @@ func (api *API) UpdateAPIShieldSchema(ctx context.Context, rc *ResourceContainer
+	// zone-level/operation-level setting
+	OverrideMitigationAction *string `json:"validation_override_mitigation_action" url:"-"`
+}
+
+// UpdateAllAPIShieldSchemaValidationSettingsParams represents the parameters to pass to update all
+// schema validation settings on the zone
+//
+// API documentation: https://developers.cloudflare.com/api/operations/api-shield-schema-validation-update-zone-level-settings
+type UpdateAllAPIShieldSchemaValidationSettingsParams struct {
+	APIShieldSchemaValidationSettings
+}
+
+// UpdateAPIShieldSchemaValidationSettingsParams represents the parameters to pass to update certain fields
+// on schema validation settings on the zone
+//
+// API documentation: https://developers.cloudflare.com/api/operations/api-shield-schema-validation-patch-zone-level-settings
+type UpdateAPIShieldSchemaValidationSettingsParams struct {
+	// DefaultMitigationAction is the mitigation to apply when there is no operation-level
+	// mitigation action defined
+	//
+	// passing a `nil` value will have no effect on this setting
+	DefaultMitigationAction *string `json:"validation_default_mitigation_action" url:"-"`
+
+	// OverrideMitigationAction when set, will apply to all requests regardless of
+	// zone-level/operation-level setting
+	//
+	// passing a `nil` value will have no effect on this setting
+	OverrideMitigationAction *string `json:"validation_override_mitigation_action" url:"-"`
+}
+
+// APIShieldSchemaValidationSettingsResponse represents the response from the GET api_gateway/settings/schema_validation endpoint.
+type APIShieldSchemaValidationSettingsResponse struct {
+	Result APIShieldSchemaValidationSettings `json:"result"`
+	Response
+}
+
+// GetAPIShieldSchemaValidationSettings retrieves zone level schema validation settings
+//
+// API documentation: https://developers.cloudflare.com/api/operations/api-shield-schema-validation-retrieve-zone-level-settings
+func (api *API) GetAPIShieldSchemaValidationSettings(ctx context.Context, rc *ResourceContainer) (*APIShieldSchemaValidationSettings, error) {
+	path := fmt.Sprintf("/zones/%s/api_gateway/settings/schema_validation", rc.Identifier)
+
+	uri := buildURI(path, nil)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var asResponse APIShieldSchemaValidationSettingsResponse
+	err = json.Unmarshal(res, &asResponse)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return &asResponse.Result, nil
+}
+
+// UpdateAllAPIShieldSchemaValidationSettings updates zone level schema validation settings
+//
+// API documentation: https://developers.cloudflare.com/api/operations/api-shield-schema-validation-update-zone-level-settings
+func (api *API) UpdateAllAPIShieldSchemaValidationSettings(ctx context.Context, rc *ResourceContainer, params UpdateAllAPIShieldSchemaValidationSettingsParams) (*APIShieldSchemaValidationSettings, error) {
```

Comment:
the way we would generally handle this is that we wouldn't include this method if the individual fields update endpoint was available (such as `UpdateAPIShieldSchemaValidationSettings`). we would opt to instead use the one method to update a single or all of the fields. does that make sense to do in this context?

Reply:
Makes sense! Have removed this in favour of the PATCH endpoint. 

https://github.com/cloudflare/cloudflare-go/compare/ce2b27fff909a073a684911518018ec0be1ff1d0..0d0a92a41e5c180aa57f6a06b77aec34c4b34764